### DISCUSSION TITLE
Expose TAA disocclusion settings in project settings

### DISCUSSION THREAD

**Discussion by izarii-dev** [godotengine/godot-proposals#10950] [3D]
The default TAA settings cause obvious smearing on movement, especially on moving foliage, I found that changing the disocclusion threshold and disocclusion scale in the taa.cpp file fixes it.

this may makes things look more aliased/noisy during movement but it's definitely more desirable to smearing in some projects.

So I suggest including it in the project settings.

**Comment by Calinou** [comment]
This was discussed when [TAA was initially being added](https://github.com/godotengine/godot/pull/61319), but it ended up not being exposed for a few reasons. I don't precisely recall why, but one of the reasons that was mentioned is that it's easy to incorrectly tweak the disocclusion factor and end up with a poor-looking result with TAA barely having any effect (which is a problem in many released games such as Monster Hunter Rise).

That said, the TAA implementation has a lot of issues such as https://github.com/godotengine/godot/issues/61905. We lack a dedicated maintainer to improve the TAA implementation, so these will take a while to be fixed already. Nowadays, FSR2 generally does a better job (you can use it at native resolution for the best possible quality). Many other engines seem to be going the same route lately, preferring dedicated upscaling solutions (that also happen to work at native resolution) over custom TAA solutions (or even custom upscalers).

**Reply by izarii-dev** [reply]
> This was discussed when https://github.com/godotengine/godot/pull/61319, but it ended up not being exposed for a few reasons. I don't precisely recall why, but one of the reasons that was mentioned is that it's easy to incorrectly tweak the disocclusion factor and end up with a poor-looking result with TAA barely having any effect (which is a problem in many released games such as Monster Hunter Rise).

Yeah, that can be a problem.

> That said, the TAA implementation has a lot of issues such as https://github.com/godotengine/godot/issues/61905. We lack a dedicated maintainer to improve the TAA implementation, so these will take a while to be fixed already. Nowadays, FSR2 generally does a better job (you can use it at native resolution for the best possible quality). Many other engines seem to be going the same route lately, preferring dedicated upscaling solutions (that also happen to work at native resolution) over custom TAA solutions (or even custom upscalers).

I find that both Godot's TAA and FSR2 have their own problems, the TAA implementation doesn't do good enough job of rejecting history (and when you adjust the disocclusion factor, the opposite problem happens) and FSR2 looks a bit unstable.

The ideal TAA solution, in my opinion, would be something similar to Epic's TSR which tend to be really good at minimizing those problems and other usual TAA problems but then again, it has relatively high performance cost which might not be well-suited for Godot.
